docs/articles/documentation/test-api/authentication/user-roles.md

```diff
@@ -168,4 +176,27 @@ test('My test', async t => {
         .useRole(role);
 ```
 
-**Default value**: `false`
\ No newline at end of file
+**Default value**: `false`
+
+## Troubleshooting
+
+### Test Actions Fail After Authentication
+
+This issue can be caused by the browser's page caching.
+
+In order to guarantee seamless test execution, browsers that run TestCafe tests should always fetch the tested page from the TestCafe proxy server. This ensures that automation scripts on the page are in sync with the server side.
+
+However, if the browser uses a cached copy of the page, automation mechanisms may be interrupted. Among other issues, this could reset authentication data in the cookies, local and session storages during navigation.
+
+If tests fail unexpectedly after authentication, try the following steps:
+
+* Enable the [preserveUrl](#optionspreserveurl) option to avoid automatic navigation.
+* If `preserveUrl` does not fix the issue, disable page caching. Note that this slows down test execution.
+
+Use the [fixture.disablePageCaching](../test-code-structure.md#disable-page-caching) and [test.disablePageCaching](../test-code-structure.md#disable-page-caching) methods to disable caching during a particular fixture or test.
+
+To disable page caching during the entire test run, use either of the following options:
```

Comment:
use a comma after each option you list below, and a period after the last option